resources/sdk/purecloudjava/templates/ApiClient.mustache

```diff
@@ -560,7 +577,12 @@ public class ApiClient implements AutoCloseable {
         ApiClientConnectorRequest connectorRequest = prepareConnectorRequest(request, isAuthRequest);
         ApiClientConnectorResponse connectorResponse = null;
         try {
-            connectorResponse = connector.invoke(connectorRequest);
+             connectorResponse = connector.invoke(connectorRequest);
```

Comment:
This line and the following if block can be removed since the do block will always be executed at least once. This will also fix a bug because the current code will always make the first retry immediately without waiting. 

Additionally, all of the request-specific properties need to be refactored out of ApiClient and into ApiClientConnectorRequest. The request object itself should be used to track those values because Java apps can be multi-threaded. Making concurrent requests would result in props like totalRetryTimeInMillis being added to by multiple requests.

On the topic of totalRetryTimeInMillis, it would be a more accurate approach to instead set a timestamp on the request right before it's made and measure the time since that timestamp to determine which bucket you're in. That provides a few benefits:

* More accurate because it won't exclude the time it takes to execute the request
* Removes the need for the variables tracking bucket stuff. You you can replace most of that logic with a check on the time since the started timestamp to see if duration < 5 mins, if < 10 mins, or > 10.
* Allows you to remove the initial check for the configured max retry time (mentioned above) because the measured running time will always be greater than 0 after the first request.

Reply:
Refactored accordingly. 
For now, I've created an inner class to make it threadsafe allowing per request/thread to create a new instance of the inner class. It can be extracted out to a separate class if necessary.